### PR TITLE
JP-3687: fix outlier_detection failures when run outside pipeline

### LINKED ISSUE #8640: outlier detection fails due to missing `array_shape` when run outside pipeline
_Issue [JP-3687](https://jira.stsci.edu/browse/JP-3687) was created on JIRA by [Brett Graham](https://jira.stsci.edu/secure/ViewProfile.jspa?name=bgraham):_

Running outlier detection outside of a pipeline with `in_memory=False` (the default) fails with:

 
```java
    raise ValueError("WCS must have array_shape attribute set.")
ValueError: WCS must have array_shape attribute set. ```
This can be seen by running `strun outlier_detection fail_short_image3_asn.json` with the association (and data) for the `run_image3_closedfile` regtest

 

<https://bytesalad.stsci.edu/ui/repos/tree/General/jwst-pipeline/dev/nircam/image/fail_short_image3_asn.json>

I expect others will similarly fail as the initial failure was found running non-regtest data shared with us from a user related to a different ticket.

Running the same command and association with:

<https://github.com/spacetelescope/jwst/commit/831124ef9e768abaca412827272bc2e6ce6f60af>

(the commit just prior to <https://github.com/spacetelescope/jwst/pull/8553)> does not fail.

The issue appears to be that the resample usage in outlier detection adds an `array_shape` to the input wcses only when the input is a ModelContainer of open models (as occurs during the pipeline run) whereas when just outlier detection is called the input to resample is a ModelContainer of filenames and the input model wcses are not modifed.

Some possible solutions are:
 * modify `compute_image_pixel_area` to handle the case of a missing array shape. It's unclear to me why both the array shape and bounding box are considered (and if there are conditions when these might differ)
 * modify outlier detection to add an array_shape to the input wcs prior to the call to `compute_image_pixel_area`

### PR DESCRIPTION
**Pull request by emolter**
Resolves [JP-3687](https://jira.stsci.edu/browse/JP-3687)

Closes #8640

This PR addresses a bug that occurs when the `outlier_detection` step is run outside the pipeline, i.e., loading the datamodel from file instead of passing it in as a `ModelContainer`.  In that case, the `wcs.array_shape` attribute sometimes does not get saved (saving of this attribute was added to gwcs within the past year, so the behavior is version-dependent). This PR sets the `wcs.array_shape` explicitly.

**Checklist for PR authors (skip items if you don't have permissions or they are not applicable)**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [ ] updated or added relevant tests
- [x] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [ ] All comments are resolved
- [ ] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Comment by emolter** [comment]
Regression tests started here, using the new GitHub Actions workflow (hope that works!) https://github.com/spacetelescope/RegressionTests/actions/runs/9940607341

**Comment by emolter** [comment]
separate regtests with same changes started here on Jenkins: https://plwishmaster.stsci.edu:8081/job/RT/job/JWST-Developers-Pull-Requests/1605/

**Comment by emolter** [comment]
all regtest failures are unrelated

File: jwst/outlier_detection/outlier_detection.py [hunk 1]
```diff
@@ -569,6 +569,7 @@ def gwcs_blot(median_model, blot_img, interp='poly5', sinscl=1.0):
     log.debug("Pixmap shape: {}".format(pixmap[:, :, 0].shape))
     log.debug("Sci shape: {}".format(blot_img.data.shape))
 
+    blot_img.meta.wcs.array_shape = blot_img.shape #needed for compute image pixel area
```
Comment (melanieclarke) [inline jwst/outlier_detection/outlier_detection.py:572]:
```suggestion
    # Set array shape, needed to compute image pixel area
    blot_img.meta.wcs.array_shape = blot_img.shape
```

**Review by melanieclarke — COMMENTED** [review]
LGTM!  Just a small comment grammar suggestion.

**Review by melanieclarke — APPROVED** [review]

**Pull request merged**